circuits/verify_signature.circom

```diff
@@ -1,27 +1,123 @@
-include "../node_modules/circomlib/circuits/eddsamimcsponge.circom";
+include "../node_modules/circomlib/circuits/compconstant.circom";
+include "../node_modules/circomlib/circuits/pointbits.circom";
+include "../node_modules/circomlib/circuits/mimcsponge.circom";
+include "../node_modules/circomlib/circuits/bitify.circom";
+include "../node_modules/circomlib/circuits/escalarmulany.circom";
+include "../node_modules/circomlib/circuits/escalarmulfix.circom";
 include "./hasher.circom";
 
+
+template EdDSAMiMCSpongeVerifier_patched() {
+  signal input Ax;
+  signal input Ay;
+
+  signal input S;
+  signal input R8x;
+  signal input R8y;
+
+  signal input M;
+
+  signal output valid;
+
+  var i;
+
+  // Ensure S<Subgroup Order
+  component snum2bits = Num2Bits(253);
+  snum2bits.in <== S;
+
+  component compConstant = CompConstant(2736030358979909402780800718157159386076813972158567259200215660948447373040);
+
+  for (i=0; i<253; i++) {
+    snum2bits.out[i] ==> compConstant.in[i];
+  }
+  compConstant.in[253] <== 0;
+  compConstant.out === 0;
```

Comment:
This should also be changed to a `no-op` style.

Reply:
Could you elaborate more on this?

Reply:
Since S is supplied by the user, it can be a malicious input that will not be smaller than the constant and will cause the proof to fail.